ocrd_utils/ocrd_utils/logging.py

```diff
@@ -79,34 +79,26 @@ def getLogger(*args, **kwargs):
 
 def initLogging():
     """
-    Sets logging defaults
+    Read logging configuration, if exists, otherwise use basicConfig
     """
-    logging.basicConfig(
-        level=logging.INFO,
-        format='%(asctime)s.%(msecs)03d %(levelname)s %(name)s - %(message)s',
-        datefmt='%H:%M:%S')
-    logging.getLogger('').setLevel(logging.INFO)
-    #  logging.getLogger('ocrd.resolver').setLevel(logging.INFO)
-    #  logging.getLogger('ocrd.resolver.download_to_directory').setLevel(logging.INFO)
-    #  logging.getLogger('ocrd.resolver.add_files_to_mets').setLevel(logging.INFO)
-    logging.getLogger('PIL').setLevel(logging.INFO)
-    # To cut back on the `Self-intersection at or near point` INFO messages
-    logging.getLogger('shapely.geos').setLevel(logging.ERROR)
-
-    # Allow overriding
 
     CONFIG_PATHS = [
         os.path.curdir,
         os.path.join(os.path.expanduser('~')),
         '/etc',
     ]
-
-
     for p in CONFIG_PATHS:
         config_file = os.path.join(p, 'ocrd_logging.conf')
         if os.path.exists(config_file):
-            logging.debug("Loading logging configuration from '%s'", config_file)
             logging.config.fileConfig(config_file)
             return
 
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(asctime)s.%(msecs)03d %(levelname)s %(name)s - %(message)s',
+        datefmt='%H:%M:%S')
+    logging.getLogger('PIL').setLevel(logging.INFO)
+    logging.getLogger('shapely.geos').setLevel(logging.ERROR)
```

Comment:
Personally, I think it is a source of coming confusions. If the configuration style is to be appreciated, why to keep programmatic relics? They smell like common quick-and-dirty solutions that shouldn't be part of an official and professional piece of software. Do you want to keep this flavor?
Remember, they will never be taken into account if a config file is present. A future user might wonder what those lines stand for and why they seem to have zero effects. 
Nevertheless, the information itself is not lost when line-drop is committed, since the repository keeps them in it's history.

Reply:
> Personally, I think it is a source of coming confusions. If the configuration style is to be appreciated, why to keep programmatic relics? They smell like common quick-and-dirty solutions that shouldn't be part of an official and professional piece of software. Do you want to keep this flavor?

You definitely need some fallback when no config file is present in the search directories. (You _could_ make that impossible by adding the source directory to the bottom of the list and adding the template config to the `package_data` directive. But that's not any less obscure IMO.)

> A future user might wonder what those lines stand for and why they seem to have zero effects.

Unlikely. A user capable of browsing the source code will also notice that it _says right here_ that this is only the fallback position.

> Nevertheless, the information itself is not lost when line-drop is committed, since the repository keeps them in it's history.

There's a huge (functional) difference between information preserved in a history and in comments: History can give you the vertical evolution of a piece of code _when you are already_ prepared to dig into it. Comments on the other hand readily avail themselves for information that is helpful to understand the intention of the present state and usually does not enter the history (for being too low-threshold / specific / contingent).